### ISSUE TITLE
type assertion "enables" re-assigning const variables

### ISSUE THREAD

**Issue by stefanpl** [microsoft/TypeScript#53582]
# Bug Report

### 🔎 Search Terms

- `type assertion const`
- `compiler assertion const`
- `Assignment to constant variable`
- `assertion assignment`
- `2588`

### 🕗 Version & Regression Information

This is the behavior in every version I tried, and I reviewed the FAQ for entries about `assertion` and `assignment`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/dev/bug-workbench/?ssl=3&ssc=1&pln=3&pc=31#code/MYewdgzgLgBBIFsCmBlKAnAlmA5jAvDAEQAWSANuSDAO4jrkAmRA3AFBsAU8yaWuMAIYQYAVzABrMCBpgAlARgBGAAwqWQA)

### 💻 Code

```ts
const someString = "hello world";

(someString as unknown) = 100;
```

### 🙁 Actual behavior

TS does **not** provide the `Cannot assign to … because it is a constant. (2588)` error.

### 🙂 Expected behavior

TS should warn me about assigning to constant variables.

**Comment by fatcerberus** [comment]
Assigning to an expression is just weird in general.  I’m inclined to think it’s a bug that `(foo as T) = bar` is even legal syntax.

**Comment by stefanpl** [comment]
Well, there's probably not much to do about a "Working as Intended". Guess I'll go looking for an eslint rule then :)
